### PR TITLE
run nuget deprecation via arg

### PR DESCRIPTION
**Pull request by mario-guerra**

### CODE REVIEW

**Comment by mario-guerra** [comment]
@weshaggard please review when you have a chance, thanks.

File: eng/scripts/Query-Azure-Packages.ps1 [hunk 1]
```diff
@@ -564,58 +569,65 @@ function Write-Nuget-Deprecated-Packages($packageList)
...
+      # Maven search api has lots of reliability issues so keeping this error
+      # handling here to keep it from failing the pipeline all the time.
+      try {
+        Write-Latest-Versions "java"
+        Write-Latest-Versions "android"
+      }
+      catch {
+        Write-Host "Exception: $_"
+        Write-Host "Maven search appears to be down currently, so
+        java and android updates might not complete successfully. See https://status.maven.org/ for current status."
```
Comment (weshaggard) [inline eng/scripts/Query-Azure-Packages.ps1:591]: Looks like an extra newline was added to this message.

**Comment by weshaggard** [comment]
@mario-guerra what is your goal with this change? Is it to run it locally for debugging? You can run the pipeline if you want to see the results.

**Comment by mario-guerra** [comment]
@weshaggard yes, the goal is local debug.

File: eng/scripts/Query-Azure-Packages.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Query-Azure-Packages.ps1:591]: Looks like this line is still split.

File: eng/scripts/Query-Azure-Packages.ps1 [hunk 2]
```diff
@@ -1,6 +1,11 @@
-[CmdletBinding()]
+[CmdletBinding(DefaultParameterSetName='UpdateVersions')]
 param (
+  [Parameter(ParameterSetName='UpdateVersions')]
   [string] $language = "all",
+
+  [Parameter(ParameterSetName='UpdateDeprecation')]
+  [switch] $updateDeprecation,
```
Comment (weshaggard) [inline eng/scripts/Query-Azure-Packages.ps1:7]: Perhaps call this parameter `$runDeprecationOnly` So it is clear that is the goal.

File: eng/scripts/Query-Azure-Packages.ps1 [hunk 3]
```diff
@@ -1,6 +1,11 @@
-[CmdletBinding()]
+[CmdletBinding(DefaultParameterSetName='UpdateVersions')]
```
Comment (weshaggard) [inline eng/scripts/Query-Azure-Packages.ps1:1]: No real need for different parameter sets. I'd suggest just adding the new parameter and doing work when that is set.

File: eng/scripts/Query-Azure-Packages.ps1 [hunk 4]
```diff
@@ -564,58 +569,65 @@ function Write-Nuget-Deprecated-Packages($packageList)
   }
 }
```
Comment (weshaggard) [inline eng/scripts/Query-Azure-Packages.ps1:571]: I would just add this special override here and exit afterwards. You won't need to change any of the other stuff. 

```
if ($runDeprecationOnly) {
  $packageList = Get-PackageListForLanguage "dotnet"
  Write-Nuget-Deprecated-Packages $packageList
  exit 0
}
```

**Comment by mario-guerra** [comment]
@weshaggard how does this version look?

Example of updated deprecation message:
https://www.nuget.org/packages/Microsoft.Azure.Management.Batch

**Review by weshaggard — APPROVED** [review]

**Pull request merged**